regression-tests.dnsdist/test_Caching.py

```diff
@@ -354,6 +354,52 @@ def testCacheDifferentCase(self):
         self.assertEquals(receivedResponse, differentCaseResponse)
 
 
+class TestTempFailureCacheTTLAction(DNSDistTest):
+
+    _config_template = """
+    pc = newPacketCache(100, 86400, 1)
+    getPool(""):setCache(pc)
+    addAction("servfail.cache.tests.powerdns.com.", TempFailureCacheTTLAction(1))
+    newServer{address="127.0.0.1:%s"}
+    """
+
+    def testTempFailureCacheTTLAction(self):
+        """
+        Cache: When a TempFailure TTL is set, it should be honored
+
+        dnsdist is configured to cache packets, plus a specific qname is
+        set up with a lower TempFailure Cache TTL. we are sending one request
+        (cache miss) and verify that the cache is hit for the following query,
+        but the TTL then expires before the larger "good" packetcache TTL.
+        """
+        name = 'servfail.cache.tests.powerdns.com.'
+        query = dns.message.make_query(name, 'AAAA', 'IN')
+        response = dns.message.make_response(query)
+        response.set_rcode(dns.rcode.SERVFAIL)
+
+        (receivedQuery, receivedResponse) = self.sendUDPQuery(query, response)
+        self.assertTrue(receivedQuery)
+        self.assertTrue(receivedResponse)
+        receivedQuery.id = query.id
+        self.assertEquals(query, receivedQuery)
+        self.assertEquals(receivedResponse, response)
+
+        # next query should hit the cache
+        (receivedQuery, receivedResponse) = self.sendUDPQuery(query, response)
```

Comment:
If you expect the query to hit the cache, I don't think you should provide a response. Otherwise it will be added to the queue and this might prove an issue later because you won't get the expected answer for the next test. It's not an issue there because you pass the exact same answer to the next test, but you might want to use:
```
self.sendUDPQuery(query, response=None, useQueue=False)
```